src/components/Receiving/ItemDetails.js

```diff
@@ -31,61 +31,85 @@ class ItemDetails extends Component {
     super(props);
 
     this.state = {
-      allChecked: false,
+      allChecked: {},
       checkedItems: [],
+      currentLine: 0,
+      lineItems: this.props.linesItemList,
     };
   }
 
   isItemChecked = (item) => (
     !!this.state.checkedItems.filter(el => el.id === item.id).length
   );
 
-  toggleAll = () => {
-    this.setState((state, props) => {
-      const allChecked = !state.allChecked;
-      const checkedItems = allChecked ? props.linesItemList : [];
+  toggleAll = (poLineId) => {
+    this.setState(({ allChecked, checkedItems, lineItems }) => {
+      const allCheckedLine = !allChecked[poLineId] || false;
+      const checkedLineItems = allCheckedLine
+        ? [...checkedItems, ...lineItems[poLineId]]
+        : checkedItems.filter(el => el.poLineId !== get(lineItems, poLineId, ''));
+
+      allChecked[poLineId] = allCheckedLine;
 
       return {
         allChecked,
-        checkedItems,
+        checkedItems: checkedLineItems,
       };
     });
   }
 
   toggleItem = (item) => {
-    this.setState(({ checkedItems }) => {
+    this.setState(({ allChecked, checkedItems }) => {
       const checkedItemList = checkedItems.filter(el => el.id !== item.id);
 
+      allChecked[item.poLineId] = false;
+
       return {
-        allChecked: false,
+        allChecked,
         checkedItems: this.isItemChecked(item) ? [...checkedItemList] : [...checkedItems, item],
       };
     });
   }
 
-  render() {
-    const { close, linesItemList, locationsOptions } = this.props;
-    const poLineNumber = get(linesItemList, [0, 'poLineNumber'], '');
-    const title = get(linesItemList, [0, 'title'], '');
-    const resultsFormatter = {
+  onClickControlButton = (type) => {
+    switch (type) {
```

Comment:
IMHO, for this kind of switch there is no too much sense, it looks like over-engineering. Just create 2 function - `clickNext` and `clickPrevious` and pass them to footer.